pkg/tenancy/tenancy.go

```diff
@@ -109,3 +111,29 @@ func GetTenantFromGRPCMetadata(ctx context.Context) (string, bool) {
 	}
 	return md.Get(DefaultTenantHeader)[0], true
 }
+
+func EnforceQueryTenancy(tenantLabel string, tenant string, query string) (string, error) {
+	labelMatcher := &labels.Matcher{
+		Name:  tenantLabel,
+		Type:  labels.MatchEqual,
+		Value: tenant,
+	}
+
+	e := injectproxy.NewEnforcer(false, labelMatcher)
+
+	expr, err := parser.ParseExpr(query)
+	if err != nil {
+		return "", errors.Errorf("error parsing query string, when enforcing tenenacy %q", err.Error())
```

Comment:
I think it wiuld be better to use `errors.Wrap()` here. 
Ideally we should also stop using this deprecated  `github.com/pkg/errors` package.